aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/CredentialStoreActions.kt

```diff
@@ -28,8 +28,8 @@ object CredentialStoreActions : StoreActions {
             logger?.verbose("$id Starting execution")
             val evt = try {
                 val credentials = legacyCredentialStore.retrieveCredential()
-                credentials?.let {
-                    credentialStore.saveCredential(it)
+                if (credentials !is AmplifyCredential.Empty) {
```

Comment:
can't this be `credentials != AmplifyCredential.Empty` as these are serializable objects.
I think `is` uses reflection.